sql/internal/specutil/convert.go

```diff
@@ -543,17 +565,52 @@ func externalRef(ref *schemaspec.Ref, sch *schema.Schema) (*schema.Table, *schem
 	return tbl, c, nil
 }
 
-func tableName(ref *schemaspec.Ref) (string, error) {
+// findTable finds the table referenced by ref in the provided schema. If the table is not in the provided
+// other schemas in the connected Realm are searched as well.
+func findTable(ref *schemaspec.Ref, sch *schema.Schema) (*schema.Table, error) {
+	qualifier, tblName, err := tableName(ref)
+	if err != nil {
+		return nil, err
+	}
+	// Search the same schema.
+	if qualifier == "" || qualifier == sch.Name {
+		tbl, ok := sch.Table(tblName)
+		if !ok {
+			return tbl, fmt.Errorf("sqlspec: table %q not found", tblName)
+		}
+		return tbl, nil
+	}
+	if sch.Realm == nil {
+		return nil, fmt.Errorf("sqlspec: table %s.%s not found", qualifier, tblName)
+	}
+	// Search for the table in another schemas in the realm.
+	sch, ok := sch.Realm.Schema(qualifier)
```

Comment:
I'll suggest always going with Realm to simplify this flow. We can talk about it if you want.